rundeckapp/grails-spa/packages/ui-trellis/src/app/components/home/HomeCardList.vue

```diff
@@ -0,0 +1,217 @@
+import HomeBrowser from "./HomeBrowser.vue";
+import {Project} from "@/app/components/home/types/projectTypes";
+import UiSocket from "@/library/components/utils/UiSocket.vue";
+import {getRundeckContext} from "@/library";
+import OffsetPagination from "@/library/components/utils/OffsetPagination.vue";
+
+export default defineComponent({
+  name: "HomeCardList",
+  components: {
+    UiSocket,
+    HomeSearchBar,
+    HomeBrowser,
+    OffsetPagination
+  },
+  props: {
+    loadedProjectNames: {
+      type: Boolean,
+      default: false
+    },
+    projects: {
+      type: Array as PropType<Project[]>,
+      default: () => []
+    }
+  },
+  data() {
+    return {
+      search: "",
+      filteredProjects: [] as Project[],
+      favoriteProjectNames: [] as Project[],
+      resultsPage: [] as Project[],
+      showSearchResults: false,
+      filterFavoritesOnly: false,
+      eventBus: getRundeckContext().eventBus,
+      pagination: {
+        offset: 0,
+        max: 30,
+        total: -1
+      }
+    }
+  },
+  computed: {
+    searchResultsCount(): number {
+      return this.filteredProjects?.length || 0
+    },
+    favoriteProjects(): Project[] {
+      if (!this.projects || !this.favoriteProjectNames) {
+        return []
+      }
+
+      return this.projects.filter((project) => this.favoriteProjectNames.includes(project.name))
+    },
+  },
+  methods: {
+    // filters the projects based on the search input and if the user has selected to only show favorites
+    // search input can either be a string or a regex, and the search is case-insensitive
+    // updates the pagination to reflect the total number of projects and sets flag to show search results message
+    // if the search input is not empty
+    handleSearch(resetPagination = true) {
+      if (resetPagination) {
+        this.pagination.offset = 0;
+      }
+
+      let regex;
+      if (this.search.charAt(0) === '/' && this.search.charAt(this.search.length - 1) === '/') {
+        regex = new RegExp(this.search.substring(1, this.search.length - 1),'i');
+      }else{
+        //simple match which is case-insensitive, escaping special regex chars
+        regex = new RegExp(this.search.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'),'i');
+      }
+
+      if(this.filterFavoritesOnly) {
+        this.filteredProjects = this.favoriteProjects.filter((project) => {
+          return project.name.match(regex) || project.label && project.label.match(regex);
+        });
+      } else {
+        this.filteredProjects = this.projects.filter((project) => {
+          return project.name.match(regex) || project.label && project.label.match(regex);
+        });
+      }
+
+      this.resultsPage = this.paginateResults(this.filteredProjects);
+
+      // update the pagination to reflect the total number of projects
+      this.pagination.total = this.searchResultsCount;
+
+      if (this.search !== "") {
+        this.showSearchResults = true;
+      }
```

Comment:
matches old behavior:
```suggestion
      if (this.search !== "" || this.filterFavoritesOnly) {
        this.showSearchResults = true;
      }else{
        this.showSearchResults = false;
      }
```